src/http/request.rs

```diff
@@ -49,7 +52,32 @@ impl<B> Request<B> {
     /// Read the Request body.
     #[inline]
     pub fn body(&self) -> Option<&B> { self.body.as_ref() }
-    
+
+    /// Take the Body of this Request
+    #[inline]
+    pub fn take_body(self) -> Option<B> { self.body }
```

Comment:
With the following implementation already existing for Response:
```
impl Response<Body> {
    /// Take the `Body` of this response.
    #[inline]
    pub fn body(mut self) -> Body {
        self.body.take().unwrap_or(Body::empty())
    }
}
```

Would this make more sense?
```
impl<B> Request<B> {...
    /// Read the Request body.
    #[inline]
    pub fn body_ref(&self) -> Option<&B> { self.body.as_ref() }
...}

impl Request<Body> {
    /// Take the Request body.
    #[inline]
    pub fn body(self) -> Body { self.body.unwrap_or_default() }
}
```

My only concern with returning a default is that I think the caller should be able to identify if a body exists or not, but `body_ref()` seems to cover that use case.

Also, should I change the above impl for Response to use `self` instead of `mut self` so it consumes the response? Or should I leave it as-is? This would reduce the differences further.